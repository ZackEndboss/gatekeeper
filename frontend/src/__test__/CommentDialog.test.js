import React from 'react';
import ConnectedCommentDialog, {CommentDialog, CommentForm}  from '../CommentDialog';
import renderer from 'react-test-renderer';
import {MockStore} from './mockStore';
import {Provider} from 'react-redux';
import {Modal, Button} from 'react-bootstrap';
import {shallow, mount} from 'enzyme';
import sinon from 'sinon';

jest.mock('../FormInputField', () => 'FormInputField');

describe('CommentDialog', () => {
    it('renders', () => {
        const wrapper = shallow(<CommentDialog group='group'
                                               service='service'
                                               environment='environment'
                                               comment='comment 1'
                                               show_comment_edit_dialog={false}/>);
        expect(wrapper.find(Modal).length).toEqual(1);
    });

    it('dialog is open', () => {
        const wrapper = shallow(<CommentDialog group='group'
                                               service='service'
                                               environment='environment'
                                               comment='comment 1'
                                               show_comment_edit_dialog={true}/>);
        expect(wrapper.find(Modal).prop('show')).toEqual(true);
    });

    it('contact form should render form correctly', () => {
        const component = renderer.create(<CommentForm comment='comment 1'/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('close button', () => {
        const buttonSpy = sinon.spy();
        const component = shallow(<CommentForm closeCommentEditDialog={buttonSpy}/>);
        component.find(Button).at(1).simulate('click');
        expect(buttonSpy.calledOnce).toBeTruthy();
    });
});

describe('load from state', () => {
    it('should load the state correctly', () => {
        const storeMock = MockStore({
            'gates': {
                'ftx': {
                    'gatekeeper': {
                        'test': {
                            group: 'ftx',
                            service: 'gatekeeper',
                            environment: 'test',
                            comment: 'some comment',
                            show_comment_edit_dialog: true
                        }
                    }
                }
            }
        });

        const subject = mount(
            <Provider store={storeMock}>
                <ConnectedCommentDialog group='ftx' service='gatekeeper' environment="test"/>
            </Provider>);
        expect(subject.find(Modal).prop('show')).toEqual(true);
    });
});