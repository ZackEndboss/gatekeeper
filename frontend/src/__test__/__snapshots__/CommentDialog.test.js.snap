// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentDialog contact form should render form correctly 1`] = `
<form
  onSubmit={undefined}
>
  <h4>
    Edit Comment
  </h4>
  <FormInputField
    inputProps={
      Object {
        "autoFocus": true,
        "component": "input",
        "type": "text",
      }
    }
    name="comment"
    placeholder="enter comment"
  />
  <button
    className="btn btn-success"
    disabled={false}
    type="submit"
  >
    Save
  </button>
  <button
    className="btn btn-default"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Close
  </button>
</form>
`;
